src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteWalletByIdTest.java

```diff
@@ -0,0 +1,66 @@
+package pl.byczazagroda.trackexpensesappbackend.integration;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import pl.byczazagroda.trackexpensesappbackend.BaseIntegrationTestIT;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransaction;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+import pl.byczazagroda.trackexpensesappbackend.model.Wallet;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionRepository;
+import pl.byczazagroda.trackexpensesappbackend.repository.WalletRepository;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+
+public class DeleteWalletByIdTest extends BaseIntegrationTestIT {
+
+    @Autowired
+    WalletRepository walletRepository;
+
+    @Autowired
+    FinancialTransactionRepository financialTransactionRepository;
+
+    @BeforeEach
+    void clearDatabase() {
+        walletRepository.deleteAll();
+    }
+
+    @DisplayName("Should delete wallet from database and return status 200")
+    @Test
+    void testDeleteWalletByIdAPI_whenWalletIdIsCorrect_thenShouldReturnAcceptAndDeleteRecord () throws Exception {
+        Wallet wallet = walletRepository.save(new Wallet("Test Wallet"));
+        createTestFinancialTransaction(wallet);
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/wallets/{id}", wallet.getId()).accept(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+        Assertions.assertEquals(0, walletRepository.count());
+
+    }
+
+    @DisplayName("Should return is Not Found error when Id does not exist")
+    @Test
+    void testDeleteWalletById_whenWalletIdIsIncorrect_thenShouldReturnNotFoundError() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/wallets/{id}", 1L).accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(MockMvcResultMatchers.status().isNotFound())
+                .andReturn();
+        Assertions.assertEquals(0, walletRepository.count());
+    }
+
+    private void createTestFinancialTransaction(Wallet wallet){
+        financialTransactionRepository.save(FinancialTransaction.builder()
+                .wallet(wallet)
+                .amount(BigDecimal.valueOf(1.0))
```

Comment:
Please use constructor for creating `BigDecimal` object with a value instead of method `.valueOf( double | float )`

**Link:**
https://stackoverflow.com/questions/7186204/bigdecimal-to-use-new-or-valueof

> BigDecimal bd1 = new BigDecimal(0.01);
> BigDecimal bd2 = new BigDecimal("0.01");
> System.out.println("bd1 = " + bd1);
> System.out.println("bd2 = " + bd2);
> 
>  // Result:
> bd1 = 0.01000000000000000020816681711721685132943093776702880859375
> bd2 = 0.01

**Also please read beloved:**
> Effective Java 3rd edition (Joshua Bloch)
> Item 60: Avoid float or double if exact answers are required
> 
> The float and double types are particularly ill-suited for monetary calculations because it is impossible to represent > 0.1 (or any other negative power of ten) as a float or double exactly.
> :
> There are, however, two disadvantages to using BigDecimal: it's a lot less convenient than using a primitive > arithmetic type, and it's a lot slower. The latter disadvantage is irrelevant if you're solving a single short problem, but > the former may annoy you.
> :
> An alternative to using BigDecimal is to use int or long, depending on the amounts involved, and to keep track of > the decimal point yourself. In this example, the obvious approach is to do all computations in cents.

Reply:
changed